### PR TITLE
Expose MaxPendingBytes on the natsOptions

### PR DESCRIPTION
**Pull request by LaurensVergote**
Similar as https://github.com/nats-io/nats.net/pull/844 from the nats.net library.
I've used the same defaults as the C# library

Couldn't get the testsuite to work locally so I hope the github pipeline will run them :)

### CODE REVIEW

**Comment by LaurensVergote** [comment]
Not quite sure why the CI timed out on one of the new tests, had no issues locally running it

File: test/test.c [hunk 1]
```diff
@@ -14407,6 +14412,71 @@ test_AsyncErrHandler(void)
...
+    IFOK(s, natsConnection_Subscribe(&sub, nc, "async_test", _recvTestString, (void*)&arg));
+
+    natsMutex_Lock(arg.m);
+    arg.sub = sub;
+    natsMutex_Unlock(arg.m);
+
+    size_t data_len = 10;
+    const char msg[] = { 0,1,2,3,4,5,6,7,8,9 }; //10 bytes long message
+
+    for (int i = 0;
```
[the remark this answers is not in the document]
Comment (kozlovic) [reply test/test.c:14453]: For code consistency. It came from a time where - I believe on Windows, with whatever compiler version we were using - it was not possible to declare variables there. We could relax that, especially in test.c, but I like to keep things consistent. It also help prevent variable shadowing bugs.

[the remark this answers is not in the document]
Comment (kozlovic) [reply test/test.c:14453]: That produces warnings:
```
[ 98%] Building C object test/CMakeFiles/testsuite.dir/test.c.o
/home/travis/build/nats-io/nats.c/test/test.c: In function ‘test_AsyncErrHandler_MaxPendingBytes’:
/home/travis/build/nats-io/nats.c/test/test.c:14453:5: warning: statement with no effect [-Wunused-value]
14453 |     for (i;
      |     ^~~
```
Instead, do `for (i=0; ...`.

File: test/test.c [hunk 2]
```diff
@@ -33526,6 +33596,89 @@ test_MicroAsyncErrorHandler(void)
...
+
+    test("Add test endpoint: ");
+    testCond(NULL == micro_add_endpoint(&ep, m, NULL, &ep_cfg, true));
+
+    natsMutex_Lock(arg.m);
+    arg.status = NATS_OK;
+    natsMutex_Unlock(arg.m);
+
+    test("Cause an error by sending too many messages: ");
+    for (i;
```
Comment (kozlovic) [inline test/test.c:33655]: Same comment than in the other test.

**Review by kozlovic — CHANGES_REQUESTED** [review]
Let's fix the for loop to not have the warning. But more importantly, we need @levb to confirm if we are still using the `dev` branch for new features. If so, this PR will need to change the base branch from `main` to `dev`. As of now though, I don't think that `dev` is up-to-date, so let's wait on @levb to confirm what we need to do here.

File: test/test.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (LaurensVergote) [reply test/test.c:14453]: Done.

File: test/test.c (hunk 2 above)
Comment (LaurensVergote) [reply test/test.c]: Done

**Review by kozlovic — APPROVED** [review]
LGTM. Will let @levb have a look and merge if he is ok with it.

**Review by levb — APPROVED** [review]
LGTM

**Pull request merged**